maestro/lifecycle.py

```diff
@@ -109,14 +109,36 @@ def _create_env(self):
         env.update(self.container_env)
         return dict((str(k), str(v)) for k, v in env.items())
 
-    def _test(self):
+    def _test(self, ship=None, container=None):
         return subprocess.call(self.command, env=self._create_env()) == 0
 
     @staticmethod
     def from_config(container, config):
         return ScriptExecutor(config['command'], container.env,
                               attempts=config.get('attempts'))
 
+class RemoteScriptExecutor(RetryingLifecycleHelper):
+    """
+    Lifecycle state helper that executes a script in side of 'Remote Container' and uses the exit code as the
+    success value.
+    """
+
+    def __init__(self, command, env, attempts):
+        RetryingLifecycleHelper.__init__(self, attempts)
+        self.command = shlex.split(command)
+        self.container_env = env
+
+    def __repr__(self):
+        return 'RemoteScriptExector({}, {} attempts)'.format(self.command, self.attempts)
+
+    def _test(self, ship, name):
+        """ Execute a script in side of 'Remote Container' """
+        return ship.backend.exec_start(ship.backend.exec_create(name, self.command)) == '0\n'
```

Comment:
I still don't understand why you can't simply check the exit status of the script, like it is done for the `exec` lifecycle check type. You'll need to use `ship.backend.exec_inspect` and make sure the script is done executing before looking at the `ExitCode` in the inspect dict.